lib/connection_string.js

```diff
@@ -500,6 +493,46 @@ function checkTLSOptions(queryString) {
   }
 }
 
+/**
+ * Checks options object if both options are present (any value) will throw an error.
+ *
+ * @param {Object} options Options Object
+ * @param {string} optionKeyA A options key
+ * @param {string} optionKeyB B options key
+ * @throws {MongoParseError}
+ */
+function assertRepelOptions(options, optionKeyA, optionKeyB) {
+  if (options.hasOwnProperty(optionKeyA) && options.hasOwnProperty(optionKeyB)) {
+    throw new MongoParseError(`The \`${optionKeyA}\` option cannot be used with \`${optionKeyB}\``);
+  }
+}
+
+/**
+ * Checks a all options for invalid tls options according to the URI options spec.
+ *
+ * @param {string} queryString The query string to check
+ * @throws {MongoParseError}
+ */
+function checkTLSOptions(options) {
+  options = options || {};
```

Comment:
[nit] If options in undef and we just make it an empty object we'll never throw an error, should we just return early?